components/shared/profile/edit/EditSocials.tsx

```diff
@@ -0,0 +1,253 @@
+import twitter from "@/public/icons/icn-twitter.svg";
+import instagram from "@/public/icons/icn-instagram.svg";
+import linkedin from "@/public/icons/icn-linkedin.svg";
+import github from "@/public/icons/icn-github.svg";
+import dribble from "@/public/icons/icn-dribbble.svg";
+import facebook from "@/public/icons/icn-facebook.svg";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { X } from "lucide-react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { SocialsSchema } from "@/lib/profileValidations";
+import Image from "next/image";
+import { Button } from "@/components/ui/button";
+import LoadingSpinner from "../../LoadingSpinner";
+import { updateUserSocials } from "@/lib/actions/user.actions";
+import { z } from "zod";
+
+interface EditSocialsProps {
+  user: Partial<IUser>;
+}
+
+const EditSocials = ({ user }: EditSocialsProps) => {
+  const [isPending, startTransition] = useTransition();
+  const { handleSubmit, register } = useForm({
+    resolver: zodResolver(SocialsSchema),
+    defaultValues: {
+      twitter: {
+        username: user.socials?.twitter?.username || "",
+        url: user.socials?.twitter?.url || "",
+      },
+      instagram: {
+        username: user.socials?.instagram?.username || "",
+        url: user.socials?.instagram?.url || "",
+      },
+      linkedin: {
+        username: user.socials?.linkedin?.username || "",
+        url: user.socials?.linkedin?.url || "",
+      },
+      github: {
+        username: user.socials?.github?.username || "",
+        url: user.socials?.github?.url || "",
+      },
+      dribbble: {
+        username: user.socials?.dribbble?.username || "",
+        url: user.socials?.dribbble?.url || "",
+      },
+      facebook: {
+        username: user.socials?.facebook?.username || "",
+        url: user.socials?.facebook?.url || "",
+      },
+    },
+  });
+  const onSubmit: SubmitHandler<z.infer<typeof SocialsSchema>> = async (
+    values
+  ) => {
+    startTransition(async () => {
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      const { twitter, instagram, linkedin, github, dribbble, facebook } =
+        values;
+      const user = await updateUserSocials({
+        twitter: {
+          username: twitter?.username ?? "",
+          url: twitter?.url ?? "",
+        },
+        instagram: {
+          username: instagram?.username ?? "",
+          url: instagram?.url ?? "",
+        },
+        linkedin: {
+          username: linkedin?.username ?? "",
+          url: linkedin?.url ?? "",
+        },
+        github: {
+          username: github?.username ?? "",
+          url: github?.url ?? "",
+        },
+        dribbble: {
+          username: dribbble?.username ?? "",
+          url: dribbble?.url ?? "",
+        },
+        facebook: {
+          username: facebook?.username ?? "",
+          url: facebook?.url ?? "",
+        },
+      });
+      if (user) {
+        window.location.reload();
```

Comment:
Let's remove the reload and once we get Toasts in just update both of these to show a toast (toast PR should take just a few mins)